src/code/z_shrink_window.c

```diff
@@ -1,25 +1,121 @@
+s32 ShrinkWindow_GetLetterboxTarget(void) {
+    return gShrinkWindowContextPtr->letterboxTarget;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_GetLetterboxMagnitude.s")
+void ShrinkWindow_SetLetterboxMagnitude(s32 magnitude) {
+    gShrinkWindowContextPtr->letterboxMagnitude = magnitude;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_SetPillarboxTarget.s")
+s32 ShrinkWindow_GetLetterboxMagnitude(void) {
+    return gShrinkWindowContextPtr->letterboxMagnitude;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_GetPillarboxTarget.s")
+void ShrinkWindow_SetPillarboxTarget(s32 target) {
+    gShrinkWindowContextPtr->pillarboxTarget = target;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_SetPillarboxMagnitude.s")
+s32 ShrinkWindow_GetPillarboxTarget(void) {
+    return gShrinkWindowContextPtr->pillarboxTarget;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_GetPillarboxMagnitude.s")
+void ShrinkWindow_SetPillarboxMagnitude(s32 magnitude) {
+    gShrinkWindowContextPtr->pillarboxMagnitude = magnitude;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_Init.s")
+s32 ShrinkWindow_GetPillarboxMagnitude(void) {
+    return gShrinkWindowContextPtr->pillarboxMagnitude;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_Fini.s")
+void ShrinkWindow_Init(void) {
+    gShrinkWindowContextPtr = &gShrinkWindowContext;
+    bzero(gShrinkWindowContextPtr, sizeof(gShrinkWindowContext));
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_Step.s")
+void ShrinkWindow_Destroy(void) {
+    gShrinkWindowContextPtr = NULL;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_shrink_window/ShrinkWindow_Draw.s")
+void ShrinkWindow_Update(s32 framerateDivisor) {
+    s32 step;
+    s32 nextMagnitude;
+
+    step = ((framerateDivisor == 3) ? 10 : 30 / framerateDivisor );
+
+    nextMagnitude = gShrinkWindowContextPtr->letterboxMagnitude;
+    Math_StepToIGet(&nextMagnitude, gShrinkWindowContextPtr->letterboxTarget, step);
+    gShrinkWindowContextPtr->letterboxMagnitude = nextMagnitude;
+    
+    nextMagnitude = gShrinkWindowContextPtr->pillarboxMagnitude;
+    Math_StepToIGet(&nextMagnitude, gShrinkWindowContextPtr->pillarboxTarget, step);
+    gShrinkWindowContextPtr->pillarboxMagnitude = nextMagnitude;
+}
+
+void ShrinkWindow_Draw(GraphicsContext *gfxCtx) {
+    Gfx* gfx;
+    s8 letterboxMagnitude = gShrinkWindowContextPtr->letterboxMagnitude;
+    s8 pillarboxMagnitude = gShrinkWindowContextPtr->pillarboxMagnitude;
+
+    if (letterboxMagnitude > 0) {
+        OPEN_DISPS(gfxCtx);
+
+        gfx = OVERLAY_DISP;
+
+        gDPPipeSync(gfx++);
+        gDPSetCycleType(gfx++, G_CYC_FILL);
+        gDPSetRenderMode(gfx++, G_RM_NOOP, G_RM_NOOP2);
+        gDPSetFillColor(gfx++, (GPACK_RGBA5551(0, 0, 0, 1) << 16) | GPACK_RGBA5551(0, 0, 0, 1));
```

Comment:
```suggestion
        gDPSetFillColor(gfx++, (GPACK_RGBA16(0, 0, 0, 1) << 16) | GPACK_RGBA16(0, 0, 0, 1));
```
Wasn't `RBGA5551` depreciated and changed to `RGBA16`. Is it possible to change this macro?

Reply:
This is a gbi.h thing, so might be a bit controversial. I suppose we could just alias it.

Reply:
I've thought about this again and think it's a bad idea, since 5551 provides strictly more information than 16. Plus I don't really want to change the gbi.h version if we can help it.

Reply:
I think I'd prefer one more third opinion on this, but I suppose its fine if you wanted to leave it. 

That same argument would apply to all `RGBA16` names as well too, although it does seem to be more of a standard name.

Reply:
I suppose the difference here is it's actually acting on individual bits that you can see, rather than just as an amorphous blob of texture data.